localstack/services/s3/s3_listener.py

```diff
@@ -686,17 +703,25 @@ def return_response(self, method, path, data, headers, response):
         if (not bucket_name or len(bucket_name) == 0) and len(hostname_parts) > 1:
             bucket_name = hostname_parts[0]
 
-        # POST requests to S3 may include a success_action_redirect field,
-        # which should be used to redirect a client to a new location.
+        # POST requests to S3 may include a success_action_redirect or
+        # success_action_status field, which should be used to redirect a
+        # client to a new location.
         key = None
         if method == 'POST':
-            key, redirect_url = multipart_content.find_multipart_redirect_url(data, headers)
+            key, redirect_url = multipart_content.find_multipart_key_value(data, headers)
 
             if key and redirect_url:
                 response.status_code = 303
                 response.headers['Location'] = expand_redirect_url(redirect_url, key, bucket_name)
                 LOGGER.debug('S3 POST {} to {}'.format(response.status_code, response.headers['Location']))
 
+            key, status_code = multipart_content.find_multipart_key_value(
+                data, headers, 'success_action_status')
+            if response.status_code == 200 and status_code == '201' and key:
```

Comment:
One more thing - Seems that `success_action_status` could also contain status codes `200` or `204` (see https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html). Shouldn't hold up the merge, but maybe we can add that as a separate PR @qasimzee ? That would be great, thanks.

Reply:
let me put something this weekend